client/src/layers/AnnotationLayers/PolygonLayer.ts

```diff
@@ -57,6 +60,32 @@ export default class PolygonLayer extends BaseLayer<PolyGeoJSData> {
       super.initialize();
     }
 
+
+    hoverAnnotations(e: GeoEvent) {
+      if (!this.drawingOther) {
+        const { found } = this.featureLayer.pointSearch(e.mouse.geo);
+        this.bus.$emit('annotation-hover', found, e.mouse.geo);
+      }
+    }
+
+    setHoverAnnotations(val: boolean) {
+      if (!this.hoverOn && val) {
+        this.featureLayer.geoOn(
+          geo.event.feature.mouseover,
+          (e: GeoEvent) => this.hoverAnnotations(e),
+        );
+        this.featureLayer.geoOn(
+          geo.event.feature.mouseoff,
+          (e: GeoEvent) => this.hoverAnnotations(e),
+        );
+        this.hoverOn = true;
+      } else if (this.hoverOn && !val) {
+        this.featureLayer.geoOff(geo.event.feature.mouseover);
+        this.featureLayer.geoOff(geo.event.feature.mouseoff);
+        this.hoverOn = false;
+      }
+    }
```

Comment:
This looks like copy/paste.  Can it be refactored into baselayer?

Reply:
I did that in the beginning but felt it may be a bit confusing with the `drawOther` requirement.  That makes it so when boxes are visible it activates on the boxe and when only polygons are visible it will activate on the polygon.  I can look for a better way to refactor it into baseLayer.

Reply:
I see.  There's a lot more copy/paste between the rectangle layer and polygon layer than I thought. Maybe they need to share a new common base class.  Probably out of scope for this PR

Reply:
+1 on that I thought about that but didn't want to rearrange too much.